### DISCUSSION TITLE
Simple local sync

### DISCUSSION THREAD

**Discussion by vnckppl** [mateusz-bak/openreads#714] [Ideas]
**Is your feature request related to a problem? Please describe.**
Several feature requests are about automated syncing. Most request propose syncing with a server. Perhaps, we could use another way of syncing the openreads database that bypasses the necesity of a central server.

**Describe the solution you'd like**
We could use syncthing to sync the openreads database between devices. From the [Syncthing](https://syncthing.net/) website: 

> Syncthing is a continuous file synchronization program. It synchronizes files between two or more computers in real time, safely protected from prying eyes. 

So, it does not require a central server, but syncs folders between two devices (P2P). Syncthing is available for Linux, Mac, Windows, and Android. I use it for example to sync my epubs. The reader I use ([Book Reader](https://gitlab.com/axet/android-book-reader)) stores meta-data files next to the epub files for keeping track of progress. Syncthing syncs these meta-data files, allowing me to keep track of my reading progress on multiple devices.

I am not sure how the openreads (binary?) database is stored, but allowing to store the database in a user defined location would allow syncing of this database to multiple devices.

**Comment by MikaelaLovett** [comment]
Agreed! ✨ This would be amazing, I use this for multiple apps/files as well and it'd make so much sense, honestly!

**Comment by mateusz-bak** [comment]
That sounds useful, so the goal is to have the database or it's mirror in user storage so it's accessible by Syncthing?

**Comment by vnckppl** [comment]
Yes exactly. It would then of course also require that the database is read entirely during startup instead of using a cached version. This would be necessary to ensure that any newly synced changes are reflected in the program.

**Comment by lee-b** [comment]
Agree, syncthing support would be great.  Especially if combined with #584 (desktop support).

**Comment by lscarso** [comment]
sync is really wellcome but what about using gdrive, iCloud or Onedrive as Db location? only concurrent access should be manage, the sync/cache/availability is managed by the OS

**Comment by meonkeys** [comment]
I run the Nextcloud app on my phone. It can be configured to sync particular folders. If openreads could write a .backup file periodically to a certain folder, the Nextcloud app could sync those. This would make for a pretty simple backup mechanism for people already using Nextcloud (or other similar mobile file sync tools).
